tests/tests_e3/python/main_test.py

```diff
@@ -103,3 +105,46 @@ def test_pypi_closure_tool():
         "src1-1.0.0-py3-none-any.whl",
         "src2-1.0.0-py3-none-any.whl",
     }
+
+
+def test_star_requirements():
+    """Test package requirements ending with * with != operator."""
+    wheel1 = generate_py_pkg_source("src1", version="1.0.4")
+    assert os.path.isfile(wheel1.path)
+    assert not wheel1.requirements
+
+    wheel2 = generate_py_pkg_source("src2", requires=["src1!=1.0.*"])
+    assert os.path.isfile(wheel2.path)
+    assert len(wheel2.requirements) == 1
+
+    wheel3 = generate_py_pkg_source("src1", version="1.1.4")
+    assert os.path.isfile(wheel3.path)
+    assert not wheel3.requirements
+
+    mkdir(".cache")
+
+    with PyPIClosure(
+        python3_version=10,
+        platforms=[
+            "x86_64-linux",
+        ],
+        cache_dir=".cache",
+        cache_file=".pypi.cache",
+    ) as pypi:
+        pypi.add_wheel(wheel1.path)
+        pypi.add_wheel(wheel2.path)
+        with pytest.raises(PyPIError, match="Cannot satisfy constraint src1!=1.0.*"):
+            pypi.add_requirement("src2==1.0.0")
+
+    with PyPIClosure(
+        python3_version=10,
+        platforms=[
+            "x86_64-linux",
+        ],
+        cache_dir=".cache",
+        cache_file=".pypi.cache",
+    ) as pypi:
+        pypi.add_wheel(wheel1.path)
+        pypi.add_wheel(wheel3.path)
```

Comment:
I may be wrong, but I would say `wheel2`, then `wheel3`. Here, `src1` is added twice (which I guess is ok), but `src2` is not part of the wheel ?

Reply:
You're right. That's fixed.